google/resource_compute_instance.go

```diff
@@ -394,12 +469,23 @@ func resourceComputeInstanceCreate(d *schema.ResourceData, meta interface{}) err
 	}
 
 	// Build up the list of disks
+
+	disks := []*compute.AttachedDisk{}
+	var bootDisk *compute.AttachedDisk
```

Comment:
We keep the `bootdisk` object in scope for the entire function but only use it to perform `nil` checks outside of the `if` block below. What if we moved it inside the `if` block, and pulled up the `ok` result of `d.GetOk("boot_disk");`, giving it a name like `hasBootDisk`?

That way, we know that code later in the function can only interact with the `bootDisk` as part of `disks`, and we can make lines like

```go
disk.Boot = i == 0 && bootDisk == nil
```

look like:

```go
disk.Boot = i == 0 && !hasBootDisk
```